realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -136,26 +136,26 @@ private RealmQuery(Realm realm, Class<E> clazz) {
         this.clazz = clazz;
         this.schema = realm.schema.getSchemaForClass(clazz);
         this.table = schema.table;
-        this.view = null;
+        this.linkView = null;
         this.query = table.where();
     }
 
     private RealmQuery(RealmResults<E> queryResults, Class<E> clazz) {
         this.realm = queryResults.realm;
         this.clazz = clazz;
         this.schema = realm.schema.getSchemaForClass(clazz);
-        this.table = queryResults.getTable();
-        this.view = null;
-        this.query = queryResults.getTable().where();
+        this.table = queryResults.getTableOrView();
+        this.linkView = null;
+        this.query = queryResults.getTableOrView().where();
```

Comment:
Do we need to execute `queryResults.getTableOrView()`?
We should use `this.table` instead if possible.